op-node/sources/fallback_client.go

```diff
@@ -0,0 +1,280 @@
+package sources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/ethereum-optimism/optimism/op-node/client"
+	"github.com/ethereum-optimism/optimism/op-node/metrics"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+	"math/big"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+// FallbackClient is an RPC client, it can automatically switch to the next l1 endpoint
```

Comment:
nit: it shouldn't care if its L1 or not L1

Reply:
removed